### PR TITLE
fix: Update `pg` support

### PR DESCRIPTION
**Pull request by matthewtusker**
Remove `pg` versions 1.2 and 1.3, add `pg` versions 1.4 and 1.5

### CODE REVIEW

**Review by arielvalentin — APPROVED** [review]

**Comment by arielvalentin** [comment]
@matthewtusker i approved this before seeing you had not signed the easy cla. 

Any chance you will be able to do that?

File: instrumentation/pg/Appraisals [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-# 1.2.3 appears to be the most popular version at RubyGems.org
-appraise 'pg-1.2' do
-  gem 'pg', '~> 1.2.3'
+# Last release 1.4.6 (February 26, 2023)
+appraise 'pg-1.4' do
+  gem 'pg', '~> 1.4'
 end
 
-# 1.3.0 significantly changed connecting and the handling of connection strings
-appraise 'pg-1.3' do
```
Comment (xuan-cao-swi) [inline instrumentation/pg/Appraisals:9]: Why dropping 1.3?

**Comment by xuan-cao-swi** [comment]
Also please take look at the [Conventional Commits Validation](https://www.conventionalcommits.org/en/v1.0.0/#summary). I think for this one, `fix:` would be good.

**Comment by robbkidd** [comment]
Per @matthewtusker's notice that the CLA signing process will be delayed for him, I've opened a different PR for the pg versions under test change: #1108.

Thank you for your help, Matthew!

**Pull request closed without merging**